daffodil-propgen/src/main/scala/org/apache/daffodil/propGen/TunableGenerator.scala

```diff
@@ -197,16 +224,53 @@ class PrimitiveTunable(name: String, schemaType: String, schemaDefault: String)
     case _ => schemaDefault
   }
 
+  private def restrictionCheck(rCheck: String, rValue: String): Option[String] = {
+    if (rValue != "") {
+      Some(s"""  if (!(v ${rCheck} ${rValue})) throwInvalidTunableValue(tunable, value)""")
+    } else {
+      None
+    }
+  }
+
+  private val minInclusive = node \\ "minInclusive" \@ "value"
```

Comment:
So it appears this is depending on the XML schema of the tunables being structured not as just an XML schema, but as one that always uses immediate simple types definitions having exactly these attributes. 

If so, you need comments in the tunables.xsd file because looking at it, first thing I'd do is factor out all the redundant type definitions which would break this scraper/generator code.